File: Tests/Testing/AngularTest/NetApplication/NetApplication.Application/Implementation/IntegrationService.cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using Intent.RoslynWeaver.Attributes;
using NetApplication.Application.Integration;
using NetApplication.Application.Interfaces;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.ServiceImplementations.ServiceImplementation", Version = "1.0")]

namespace NetApplication.Application.Implementation
{
    [IntentManaged(Mode.Merge)]
    public class IntegrationService : IIntegrationService
    {
        public const string ReferenceNumber = "refnumber_1234";
        public const string DefaultString = "string value";
        public const int DefaultInt = 55;
        public const string ExceptionMessage = "Some exception message";
        public static readonly Guid DefaultGuid = Guid.Parse("b7698947-5237-4686-9571-442335426771");
        public const string Param1Value = "param 1";
        public const int Param2Value = 42;

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public IntegrationService()
        {
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<CustomDTO> QueryParamOp(string param1, int param2, CancellationToken cancellationToken = default)
        {
            if (param1 != Param1Value)
            {
                throw new ArgumentException($@"{nameof(param1)} is not ""{Param1Value}"" but is ""{param1}""");
            }
            if (param2 != Param2Value)
            {
                throw new ArgumentException($@"{nameof(param2)} is not ""{Param2Value}"" but is ""{param2}""");
            }
            return CustomDTO.Create(ReferenceNumber);
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task HeaderParamOp(string param1, CancellationToken cancellationToken = default)
        {
            if (param1 != Param1Value)
            {
                throw new ArgumentException($@"{nameof(param1)} is not ""{Param1Value}"" but is ""{param1}""");
            }
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task RouteParamOp(string param1, CancellationToken cancellationToken = default)
        {
            if (param1 != Param1Value)
            {
                throw new ArgumentException($@"{nameof(param1)} is not ""{Param1Value}"" but is ""{param1}""");
            }
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task BodyParamOp(CustomDTO param1, CancellationToken cancellationToken = default)
        {
            if (param1.ReferenceNumber != ReferenceNumber)
            {
                throw new ArgumentException($@"{nameof(param1.ReferenceNumber)} is not ""{ReferenceNumber}"" but is ""{param1.ReferenceNumber}""");
            }
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task ThrowsException(CancellationToken cancellationToken = default)
        {
            throw new Exception(ExceptionMessage);
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<Guid> GetWrappedPrimitiveGuid(CancellationToken cancellationToken = default)
        {
            return DefaultGuid;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<string> GetWrappedPrimitiveString(CancellationToken cancellationToken = default)
        {
            return DefaultString;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<int> GetWrappedPrimitiveInt(CancellationToken cancellationToken = default)
        {
            return DefaultInt;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<Guid> GetPrimitiveGuid(CancellationToken cancellationToken = default)
        {
            return DefaultGuid;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<string> GetPrimitiveString(CancellationToken cancellationToken = default)
        {
            return DefaultString;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<int> GetPrimitiveInt(CancellationToken cancellationToken = default)
        {
            return DefaultInt;
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<List<string>> GetPrimitiveStringList(CancellationToken cancellationToken = default)
        {
            return new List<string> { DefaultString };
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task NonHttpSettingsOperation(CancellationToken cancellationToken = default)
        {
        }

        [IntentManaged(Mode.Fully, Body = Mode.Ignore)]
        public async Task<CustomDTO> GetInvoiceOpWithReturnTypeWrapped(CancellationToken cancellationToken = default)
        {
            return CustomDTO.Create(ReferenceNumber);
        }

        public void Dispose()
        {
        }
    }
}

File: Tests/Testing/AngularTest/NestJSApplication/src/services/integration.service.ts
import { Injectable } from '@nestjs/common';
import { CustomDTO } from './dto/Integration/custom.dto';
import { IntentIgnoreBody, IntentIgnore } from './../intent/intent.decorators';

@Injectable()
export class IntegrationService {

  @IntentIgnore()
  public readonly ReferenceNumber: string = "refnumber_1234";
  @IntentIgnore()
  public readonly DefaultString: string = "string value";
  @IntentIgnore()
  public readonly DefaultInt: number = 55;
  @IntentIgnore()
  public readonly ExceptionMessage: string = "Some exception message";
  @IntentIgnore()
  public readonly DefaultGuid: string = "b7698947-5237-4686-9571-442335426771";
  @IntentIgnore()
  public readonly Param1Value: string = "param 1";
  @IntentIgnore()
  public readonly Param2Value: number = 42;

  //@IntentCanAdd()
  constructor() {}

  @IntentIgnoreBody()
  async queryParamOp(param1: string, param2: number): Promise<CustomDTO> {
    if (param1 != this.Param1Value) {
      throw new Error(`${"param1"} is not "${this.Param1Value}" but is "${param1}"`);
    }
    if (param2 != this.Param2Value) {
      throw new Error(`${"param2"} is not "${this.Param2Value}" but is "${param2}"`);
    }
    return {
      referenceNumber: this.ReferenceNumber
    };
  }

  @IntentIgnoreBody()
  async headerParamOp(param1: string): Promise<void> {
    if (param1 != this.Param1Value) {
      throw new Error(`${"param1"} is not "${this.Param1Value}" but is "${param1}"`);
    }
  }

  @IntentIgnoreBody()
  async routeParamOp(param1: string): Promise<void> {
    if (param1 != this.Param1Value) {
      throw new Error(`${"param1"} is not "${this.Param1Value}" but is "${param1}"`);
    }
  }

  @IntentIgnoreBody()
  async bodyParamOp(param1: CustomDTO): Promise<void> {
    if (param1.referenceNumber != this.ReferenceNumber) {
      throw new Error(`${"param1.referenceNumber"} is not "${this.ReferenceNumber}" but is "${param1.referenceNumber}"`);
    }
  }

  @IntentIgnoreBody()
  async throwsException(): Promise<void> {
    throw new Error(this.ExceptionMessage);
  }

  @IntentIgnoreBody()
  async getWrappedPrimitiveGuid(): Promise<string> {
    return this.DefaultGuid;
  }

  @IntentIgnoreBody()
  async getWrappedPrimitiveString(): Promise<string> {
    return this.DefaultString;
  }

  @IntentIgnoreBody()
  async getWrappedPrimitiveInt(): Promise<number> {
    return this.DefaultInt;
  }

  @IntentIgnoreBody()
  async getPrimitiveGuid(): Promise<string> {
    return this.DefaultGuid;
  }

  @IntentIgnoreBody()
  async getPrimitiveString(): Promise<string> {
    return this.DefaultString;
  }

  @IntentIgnoreBody()
  async getPrimitiveInt(): Promise<number> {
    return this.DefaultInt;
  }

  @IntentIgnoreBody()
  async getPrimitiveStringList(): Promise<string[]> {
    return [this.DefaultString];
  }

  @IntentIgnoreBody()
  async nonHttpSettingsOperation(): Promise<void> {
    
  }

  @IntentIgnoreBody()
  async getInvoiceOpWithReturnTypeWrapped(): Promise<CustomDTO> {
    return {
      referenceNumber: this.ReferenceNumber
    };
  }
}
